src/lib/ngsiNotify/Notifier.cpp

```diff
@@ -50,32 +50,302 @@ Notifier::~Notifier (void)
+  const std::vector<std::string>&  attrsOrder
+)
+{
+  Verb                                verb = httpInfo.verb;
+  std::string                         method;
+  std::string                         url;
+  std::string                         payload;
+  std::map<std::string, std::string>  qs;
+  std::map<std::string, std::string>  headers;
+
+  LM_W(("KZ: In templateNotify"));
+  //
+  // 1. Verb/Method
+  //
+  if (verb == NOVERB)
+  {
+    // Default verb/method is POST
+    verb = POST;
+  }
+  method = verbName(verb);
+
+
+  //
+  // 2. URL
+  //
+  substitute(&url, httpInfo.url, ce);
+
+
+  //
+  // 3. Payload
+  //
+  if (httpInfo.payload == "")
+  {
+    payload = "{" + ce.toJson(renderFormat, attrsOrder) + "}";
+    LM_W(("KZ: payload: %s", payload.c_str()));
+  }
+  else
+  {
+    substitute(&payload, httpInfo.payload, ce);
+  }
+
+
+  //
+  // 4. URI Params (Query Strings)
+  //
+  for (std::map<std::string, std::string>::const_iterator it = httpInfo.qs.begin(); it != httpInfo.qs.end(); ++it)
+  {
+    std::string key   = it->first;
+    std::string value = it->second;
+
+    substitute(&value, it->second, ce);
+    qs[key] = value;
+  }
+
+
+  //
+  // 5. HTTP Headers
+  //
+  for (std::map<std::string, std::string>::const_iterator it = httpInfo.headers.begin(); it != httpInfo.headers.end(); ++it)
+  {
+    std::string key   = it->first;
+    std::string value = it->second;
+
+    substitute(&value, it->second, ce);
+    headers[key] = value;
+  }
+
+
+  //
+  // 6. Split URI in protocol, host, port and path
+  //
+  std::string  protocol;
+  std::string  host;
+  int          port;
+  std::string  uriPath;
+
+  if (!parseUrl(url, host, port, uriPath, protocol))
+  {
+    LM_E(("Runtime Error (not sending NotifyContextRequest: malformed URL: '%s')", httpInfo.url.c_str()));
+    return false;
+  }
+
+
+  //
+  // 7. Add URI params from template to uriPath
+  //
+  std::string  uri = uriPath;
+
+  if (headers.size() != 0)
+  {
+    uri += "?";
+
+    int ix = 0;
+    for (std::map<std::string, std::string>::iterator it = qs.begin(); it != qs.end(); ++it)
+    {
+      if (ix != 0)
+      {
+        uri += "&";
+      }
+
+      uri += it->first + "=" + it->second;
+      ++ix;
+    }
+  }
+
+
+  //
+  // 8. HTTP headers - to httpRequestSend via parameter 'extraHeaders'
+  //
+  std::string  out;
+  int          r;
+
+  LM_W(("KZ: Sending notification: %s [%s]", uri.c_str(), payload.c_str()));
+  r = httpRequestSend(host,
+                      port,
+                      protocol,
+                      method,
+                      tenant,
+                      ce.entityId.servicePath,
+                      xauthToken,
+                      uri,
+                      "application/json",
+                      payload,
+                      fiwareCorrelator,
+                      "",                  // ngisv2AttrFormat
+                      false,               // don't use rush ... ?
```

Comment:
I really didn't know what to put here and I didn't spend any time looking into it.
Let's do that now ...

From httpRequestSend:

    //
    // Rush
    // Every call to httpRequestSend specifies whether RUSH should be used or not.
    // But, this depends also on how the broker was started, so here the 'useRush'
    // parameter is cancelled in case the broker was started without rush.
    //
    // If rush is to be used, the IP/port is stored in rushHeaderIP/rushHeaderPort and
    // after that, the host and port of rush is set as ip/port for the message, that is
    // now sent to rush instead of to its final destination.
    // Also, a few HTTP headers for rush must be setup.
    //
    if ((rushPort == 0) || (rushHost == ""))
    {
      useRush = false;
    }
    
    if (useRush)
    {
      ...

So, if we put a `true` for useRush, then Rush will be used only if the CLI arg tells the broker to use Rush.

Using `useRush = false` would provoke rush to **never** be used for the notification.
A value of `true` seems like the better alternative, right?